test/array.cpp

```diff
@@ -454,3 +454,28 @@ TEST(Device, unequal)
         ASSERT_EQ(ptr, b.device<float>());
     }
 }
+
+TEST(DeviceId, Same)
+{
+    array a = randu(5,5);
+    ASSERT_EQ(getDevice(), getDeviceId(a));
+}
+
+TEST(DeviceId, Different)
+{
+    if (getDeviceCount() < 2) return;
+    array a = randu(5,5);
+    ASSERT_EQ(getDeviceId(a), 0);
```

Comment:
This should not be asserted without doing an explicit `setDevice`. It is possible to run tests on other devices.